src/Packetery/Module/Order/Repository.php

```diff
@@ -157,39 +157,40 @@ public function processPostClauses( array $clauses, \WP_Query $queryObject, arra
 	/**
 	 * Create table to store orders.
 	 *
-	 * @return bool
+	 * @return void
 	 */
-	public function createTable(): bool {
-		return $this->wpdbAdapter->query(
-			'CREATE TABLE IF NOT EXISTS `' . $this->wpdbAdapter->packetery_order . '` (
-				`id` bigint(20) unsigned NOT NULL,
-				`carrier_id` varchar(255) NOT NULL,
-				`is_exported` boolean NOT NULL,
-				`packet_id` varchar(255) NULL,
-				`is_label_printed` boolean NOT NULL,
-				`point_id` varchar(255) NULL,
-				`point_name` varchar(255) NULL,
-				`point_url` varchar(255) NULL,
-				`point_street` varchar(255) NULL,
-				`point_zip` varchar(255) NULL,
-				`point_city` varchar(255) NULL,
-				`address_validated` boolean NOT NULL,
-				`delivery_address` TEXT NULL,
-				`weight` float NULL,
-				`length` float NULL,
-				`width` float NULL,
-				`height` float NULL,
-				`adult_content` boolean NULL,
-				`value` double NULL,
-				`cod` double NULL,
-				`api_error_message` text NULL,
-				`api_error_date` datetime NULL,
-				`carrier_number` varchar(255) NULL,
-				`packet_status` varchar(255) NULL,
-				`deliver_on` date NULL,
-				PRIMARY KEY (`id`)
-			) ' . $this->wpdbAdapter->get_charset_collate()
-		);
+	public function createOrAlterTable(): void {
+		$createTableQuery = 'CREATE TABLE ' . $this->wpdbAdapter->packetery_order . ' (
+			`id` bigint(20) unsigned NOT NULL,
+			`carrier_id` varchar(255) NOT NULL,
+			`is_exported` tinyint(1) NOT NULL,
+			`packet_id` varchar(255) NULL,
+			`is_label_printed` tinyint(1) NOT NULL,
+			`point_id` varchar(255) NULL,
+			`point_name` varchar(255) NULL,
+			`point_url` varchar(255) NULL,
+			`point_street` varchar(255) NULL,
+			`point_zip` varchar(255) NULL,
+			`point_city` varchar(255) NULL,
+			`address_validated` tinyint(1) NOT NULL,
+			`delivery_address` text NULL,
+			`weight` float NULL,
+			`length` float NULL,
+			`width` float NULL,
+			`height` float NULL,
+			`adult_content` tinyint(1) NULL,
+			`value` double NULL,
+			`cod` double NULL,
+			`api_error_message` text NULL,
+			`api_error_date` datetime NULL,
+			`carrier_number` varchar(255) NULL,
+			`packet_status` varchar(255) NULL,
+			`deliver_on` date NULL,
+			PRIMARY KEY  (`id`)
+		) ' . $this->wpdbAdapter->get_charset_collate();
+
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		$this->wpdbAdapter->logDbDeltaResult( dbDelta( $createTableQuery ) );
```

Comment:
To že jsi do WpdbAdapteru dal metodu log* jsi z něj udělal Logger.

Volání dbDelta() + logování by se dle mne mělo dělat v Upgrade třídě.

Reply:
Adapter již obsahuje logovací metody, které jsi sám napsal. Jeho smysl je, že rozšiřuje možnosti databázové vrstvy WP a usnadňuje její použití v prostředí našeho pluginu. V tomto ohledu mi přijde jako nevhodnější místo, kam umístit zpracování výsledku databázové funkce, které je potřeba provést po každém jejím zavolání. Kód jsem upravil, aby tento záměr byl zřetelnější a kód se zjednodušil.